Use master jenkins shared pipeline

ameenhaq opened the pull request:
Updated the public library to master

ameenhaq committed:
Use Jenkins shared pipeline@master

Switch from develop to master version of the shared pipeline code.

REG-426

willis7 reviewed (changes requested):
"Updated Jenkinsfile" isn't useful as a commit message - git tells me that already "Use master jenkins shared pipeline" is way more informative.

willis7 commented:
@ameenhaq I was referring to the commit message not the title of the PR 😂

willis7 commented:
I updated the commit message which I was required to force push as you were on leave.

willis7 committed:
Add isBranch expression to Jenkinsfile

The `when branch` clause within Jenkins doesn't work. Instead, implement a isBranch
function that will assert the build is using a specific branch.
